storage/reads/array_cursor.gen.go.tmpldata

```diff
@@ -5,19 +5,37 @@
 		"Type":"float64",
 		"Aggs": [
 			{
-		 		"Name":"Count",
-				"AccType":"int64",
-				"AccTypeName":"Integer",
-		 		"AccInit":"0",
-				"Accumulate":"acc++"
-		 	},
+				"Name":"Count",
+				"OutputTypeName":"Integer",
+				"AccDecls":"var acc int64 = 0",
+				"Accumulate":"acc++",
+				"AccEmit": "c.res.Timestamps[pos] = windowEnd; c.res.Values[pos] = acc",
+				"AccReset":"acc = 0"
+			},
 			{
-		 		"Name":"Sum",
-				"AccType":"float64",
-				"AccTypeName":"Float",
-		 		"AccInit":"0",
-				"Accumulate":"acc += a.Values[rowIdx]"
-		 	}
+				"Name":"Sum",
+				"OutputTypeName":"Float",
+				"AccDecls":"var acc float64 = 0",
+				"Accumulate":"acc += a.Values[rowIdx]",
+				"AccEmit":"c.res.Timestamps[pos] = windowEnd; c.res.Values[pos] = acc",
+				"AccReset":"acc = 0"
+			},
+			{
+				"Name":"Min",
```

Comment:
Is there any overlap between first and min? This template seems very complex to me because it contains larger amounts of code.

A big concern of mine is that these code snippets appear to rely on local variables and be a way to insert code rather than as a data-driven way to instantiate templates.

Reply:
The execution pattern of first/last and min/max is pretty different, because for first/last, we do not need to read all of the blocks in a series. So Josh wrote custom cursors for them.

> A big concern of mine is that these code snippets appear to rely on local variables and be a way to insert code rather than as a data-driven way to instantiate templates.

I agree that it's not ideal. Maybe when we have generics in Go it will be easier. In the meantime, can you explain what you mean by "a data-driven way to instantiate templates"?

Reply:
I mostly mean only using attributes that would be valid if generics in Go existed. Which would limit what could be in the template to values and types rather than code.

But it's not clear to me if that would make things more clear in this example so it's a very vague idea.